<launch>
 <param name="use_sim_time" value="true" />
 <node pkg="tf" type="static_transform_publisher" name="laser_broadcaster" args="0 0 0.02 0 0 0 chassis hokuyo 100" />
 <node pkg="tf" type="static_transform_publisher" name="camera_publisher" args="0.05 0 0 0 0 0 chassis camera 100" />
 <node pkg="gmapping" type="slam_gmapping" name="gmapping_node" output="log">
        <param name="base_frame" value="chassis"/>
        <param name="odom_frame" value="odom" />
        <param name="delta" value="0.01" />
        <param name="xmin" value="-5" />
        <param name="ymin" value="-5" />
        <param name="xmax" value="5" />
        <param name="ymax" value="5" />
        <param name="maxUrange" value="5" />
        <param name="map_update_interval" value="1" />
        <param name="linearUpdate" value="0.05" />
        <param name="angularUpdate" value="0.05" />
        <param name="temporalUpdate" value="0.1" />
        <param name="particles" value="100" />
 </node>
    <!--node pkg="mybot_navigation" type="recognizer_node.py" name="recognizer" output="screen"/-->

 <node pkg="amcl" type="amcl" name="amcl" output="screen">
      <remap from="scan" to="mybot/laser/scan"/>
      <param name="odom_frame_id" value="odom"/>
      <param name="odom_model_type" value="diff-corrected"/>
      <param name="base_frame_id" value="chassis"/>
      <param name="update_min_d" value="0.5"/>
      <param name="update_min_a" value="1.0"/>  
 </node>

 <node pkg="move_base" type="move_base" name="move_base" output="log">
        <param name="base_global_planner" value="navfn/NavfnROS"/>
        <param name="controller_frequency" value="10.0" />
        <rosparam file="$(find mybot_navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find mybot_navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find mybot_navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find mybot_navigation)/config/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find mybot_navigation)/config/trajectory_planner.yaml" command="load" />
 </node>
    
 <node pkg="explore_lite" type="explore" respawn="true" name="explore" output="screen">
        <rosparam file="$(find mybot_navigation)/config/exploration.yaml" command="load" />
 </node>

    <!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find mybot_description)/rviz/frontiers.rviz"/-->
    <!--replace above with valid rviz file/-->
    <include file="$(find mybot_navigation)/launch/mybot_object_recognition.launch"/>
</launch>

